### PR TITLE
feat: VCFAnnotator accepts both seqrepo dps + clean up

### LINKED ISSUE #122: VCF / VRS Annotation should allow either local or rest seqrepo dp
Currently, it defaults to `SeqRepoDataProxy(SeqRepo("/usr/local/share/seqrepo/latest"))`, however we should allow users to choose between rest or local. We should also allow them to provide the path/url if not using the default values.

### PR DESCRIPTION
**Pull request by korikuzma**
Close #122

Notes:
- Replaces argparse with click
- Allows users to select from either local or REST SeqRepo Data Proxy
- Cleans up the VCFAnnotator code

### CODE REVIEW

File: src/ga4gh/vrs/extras/vcf_annotation.py [hunk 1]
```diff
@@ -1,122 +1,168 @@
...
+@click.option(
+    "--seqrepo_base_url",
+    required=False,
+    default="http://localhost:5000/seqrepo",
+    help="The base url for SeqRepo REST API"
+)
+def annotate_click(vcf_in: str, vcf_out: str, vrs_file: str,
+                   seqrepo_dp_type: SeqRepoProxyType, seqrepo_root_dir: str,
+                   seqrepo_base_url: str) -> None:
+    """Annotate VCF file via click"""
```
Comment (jsstevenson) [inline src/ga4gh/vrs/extras/vcf_annotation.py:64]: I think retaining the example command (`ex. python3 src/ga4gh/vrs/extras/vcf_annotation.py input.vcf.gz --out ./output.vcf.gz --vrs-file ./vrs_objects.pkl` from the previous commit) might be worthwhile -- you add a new line in here

Comment (korikuzma) [reply src/ga4gh/vrs/extras/vcf_annotation.py:64]: Probably would be good to include a README for the tool once it’s complete. Can do in #121

Comment (jsstevenson) [reply src/ga4gh/vrs/extras/vcf_annotation.py:64]: Is that a findable location? I don't know if I'd expect to find an execution example in the `if __name__...` clause. If you put it in the docstring, then it's accessible via the `--help` message, but readme makes sense as well

Comment (korikuzma) [reply src/ga4gh/vrs/extras/vcf_annotation.py]: Ended updating in this branch since #121 will not be addressed in the project.

**Review by wesleygoar — APPROVED** [review]

**Review by ahwagner — APPROVED** [review]

**Pull request merged**